### ISSUE TITLE
Window resize slow and choppy

### ISSUE THREAD

**Issue by MartinX3** [sonnyp/Tangram#192]
**Tangram version**
1.4.2

**Linux distribution**
Arch Linux

**Flatpak version**
N/A

**Desktop environment**
XFCE 4.16 with X-ORG

The maximazing, normalizing and resizing of the window is slow and choppy.
I see one of the WebKitWebProcess's eating my CPU while doing that.

Sadly I don't know how to get logs about this behavior.

I don't have this in Gnome Web. Or maybe it's there too, but Tangram requests more CPU calculations in this process?

**Comment by sonnyp** [comment]
Could you try with Tangram 2.0 beta?

Instructions here here https://thisweek.gnome.org/posts/2023/01/twig-77/#tangram

**Comment by MartinX3** [comment]
Thank you for your hard work.
I just need to wait for https://aur.archlinux.org/packages/tangram to upgrade it to 1.5.0
I hope I'm not affected by the nvidia issue.

**Comment by sonnyp** [comment]
I doubt 1.5.0 will fix this window resize performance issue, but you can try.
The nvidia issue is on 2.0 beta only.

**Comment by MartinX3** [comment]
Yeah, 1.5.0 is still choppy and the youtube video is accelerated on the CPU.
Compared to chromium it is visible slower.

Now I need to wait for this package to get updated to 2.0.0. :)

**Comment by sonnyp** [comment]
Tangram 2.0 beta is available to test from flathub https://thisweek.gnome.org/posts/2023/01/twig-77/#tangram

It you want to improve chances that it's solved for the stable release – please give it a try.

**Comment by MartinX3** [comment]
Sadly I can't install it because of `Fehler: The application re.sonny.Tangram/x86_64/beta requires the runtime org.gnome.Platform/x86_64/43 which was not found`
(I personally prefer AUR builds and dislike flatpak and similliar stuff, I only container on my servers)

**Comment by sonnyp** [comment]
```
flatpak install org.gnome.Platform//43
```

**Comment by MartinX3** [comment]
Thank you
No nvidia issue, so it uses my iGPU instead.
Also it looks prettier now.

But compared to chromium the resize and maximize minimize of the window is still slower.
Also the CPU and GPU usage is higher compared to chromium.
I test it while watching a 1080p youtube video.

It's weird, that tangram uses much more `Render/3D` on my intel iGPU and uses much more of my cpu threads while watching the 1080p youtube video in fullscreen compared to chromium.

**Comment by sonnyp** [comment]
You originally ticket said

> I don't have this in Gnome Web.

Which is why I considered it valid.

I can't do anything about performance delta with chromium.

Is Tangram resize experience now comparable with GNOME Web ?

**Comment by MartinX3** [comment]
Ah, sorry, I forgot to compare it with epiphany.
I tested it with v43 now.
Sadly the gnome browser (apple safari engine) is worse compared to firefox and chromium.
It's choppy and blinking on resizing and uses more gpu and cpu.

**Comment by sonnyp** [comment]
Apologies – I thought this was rendering related, but actually this is causing the window resize choppiness:

```
  settings.bind(
    "window-width",
    window,
    "default-width",
    Gio.SettingsBindFlags.DEFAULT,
  );
  settings.bind(
    "window-height",
    window,
    "default-height",
    Gio.SettingsBindFlags.DEFAULT,
  );
```

I will fix it for 2.0

**Comment by sonnyp** [comment]
Some thoughts/findings here https://gitlab.gnome.org/Teams/documentation/developer-www/-/issues/49
